src/main/java/decodes/cwms/algo/ResEvapAlgo.java

```diff
@@ -0,0 +1,526 @@
+			//log.warn("No existing time series. Will attempt to create.");
+			try {
+				TimeSeriesIdentifier tsId = tsdb.makeEmptyTsId();
+				tsId.setUniqueString(tsIdStr);
+				Site site = tsdb.getSiteById(siteDAO.lookupSiteID(tsId.getSiteName()));
+				if (site == null) {
+					site = new Site();
+					site.addName(new SiteName(site, Constants.snt_CWMS, tsId.getSiteName()));
+					siteDAO.writeSite(site);
+				}
+				tsId.setSite(site);
+				//log.info("Calling createTimeSeries");
+				timeSeriesDAO.createTimeSeries(tsId);
+				//log.info("After createTimeSeries, ts key = {}", tsId.getKey());
+				return tsId;
+			} catch (Exception ex2) {
+				throw new DbIoException(String.format("No such time series and cannot create for '%'", tsIdStr), ex);
+			}
+		}
+	}
+
+	//TODO read database
+//	public double getStartElevation(){
+//		return 0;
+//	}
+
+	public double[] getProfiles() throws Exception {
+		Date LastTime = HourlyEvapTS.findPrev(_timeSliceBaseTime).getTime();
+		hourlyWTP = new WaterTempProfiles(timeSeriesDAO, WtpTsid, LastTime, _timeSliceBaseTime, start_depth, depth_increment);
+		double[] arrayWTP = new double[hourlyWTP.size()];
+		for(int i = 0; i < hourlyWTP.size(); i++){
+			try {
+				arrayWTP[i] = hourlyWTP.getTimeSeriesAt(i).findPrev(_timeSliceBaseTime).getDoubleValue();
+			}
+			catch (Exception ex){
+				throw new Exception("failed to load data from WTP");
+			}
+		}
+		return arrayWTP;
+	}
+
+	public void setProfiles(WaterTempProfiles newWTP, double[] wtp, Date CurrentTime) throws DbCompException {
+		double currentDepth = start_depth;
+		for(int i = 0; i < wtp.length; i++){
+			if(i+1>newWTP.size()){
+				try{
+					TimeSeriesIdentifier newTSID = makeTSID(WtpTsid+currentDepth);
+					CTimeSeries CTProfile = new CTimeSeries(newTSID);
+					CTProfile.addSample(new TimedVariable(new Variable(wtp[i]), CurrentTime));
+					newWTP.addTimeSeries(CTProfile);
+				}
+					catch (Exception ex){
+					throw new DbCompException("failed to create new timeSeriesID"+ex);
+				}
+			}
+			else{
+				// todo might need to change
+				CTimeSeries CTProfile = newWTP.getTimeSeriesAt(i);
+				CTProfile.addSample(new TimedVariable(new Variable(wtp[i]), CurrentTime));
+			}
+			currentDepth += depth_increment;
+		}
+	}
+
+	public void setDailyProfiles(Date CurrentTime) throws DbCompException {
+		double[] arrayWTP = new double[hourlyWTP.size()];
+		int i = 0;
+		Date newTime = CurrentTime;
+		for(CTimeSeries CTS: hourlyWTP.getAllTimeSeries()){
+			try{
+				arrayWTP[i] = CTS.findPrev(CurrentTime).getDoubleValue();
+				newTime = CTS.findPrev(CurrentTime).getTime();
+			}catch(NoConversionException ex){
+				throw new DbCompException("Failed to load value from timeseries"+ex);
+			}
+			i++;
+		}
+		setProfiles(DailyWTP, arrayWTP, newTime);
+	}
+
+	public void calcDailyEvap(NamedVariable output, CTimeSeries tsc,  Date CurrentTime) throws DbCompException, NoConversionException, DecodesException {
+		double TotalEvap = calcDailyACC(tsc, CurrentTime);
```

Comment:
since moving to aggregating algorithm you can just keep a running sum of the evap for the day, along with the number of values added to the sum.